test/plots/linePlotTests.ts

```diff
@@ -683,4 +683,157 @@ describe("Plots", () => {
 
     });
   });
+
+  describe("Line Plot", () => {
+    describe("Cropped Rendering Performance", () => {
+
+      let svg: d3.Selection<void>;
+      let plot: Plottable.Plots.Line<number>;
+
+      let xScale: Plottable.Scales.Linear;
+      let yScale: Plottable.Scales.Linear;
+
+      let SVG_WIDTH = 400;
+      let SVG_HEIGHT = 400;
+
+      beforeEach(() => {
+        svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
+        xScale = new Plottable.Scales.Linear();
+        yScale = new Plottable.Scales.Linear();
+        plot = new Plottable.Plots.Line<number>();
+        plot.x((d) => d.x, xScale).y((d) => d.y, yScale);
+      });
+
+      it("can set the croppedRendering option", () => {
+        plot.renderTo(svg);
+
+        assert.isFalse(plot.croppedRenderingEnabled(), "croppedRendering is not enabled by default");
+
+        assert.strictEqual(plot.croppedRenderingEnabled(true), plot, "enabling the croppedRendering option returns the plot");
+        assert.isTrue(plot.croppedRenderingEnabled(), "can enable the croppedRendering option");
+
+        plot.croppedRenderingEnabled(false);
+        assert.isFalse(plot.croppedRenderingEnabled(), "can disable the croppedRendering option");
+
+        svg.remove();
+      });
+
+      it("does not render lines that are outside the viewport", () => {
+        let data = [
+          {x: 1, y: 1},
+          {x: 2, y: 2},
+          {x: 3, y: 1},
+          {x: 4, y: 2},
+          {x: 5, y: 1}
+        ];
+        plot.addDataset(new Plottable.Dataset(data));
+
+        // Only middle point is in viewport
+        xScale.domain([2.5, 3.5]);
+
+        plot.croppedRenderingEnabled(true);
+        plot.renderTo(svg);
+
+        let path = plot.content().select("path.line").attr("d");
+        let expectedRenderedData = [1, 2, 3].map((d) => data[d]);
```

Comment:
Instead of using these magic numbers, can we tie these values together more cleanly?  For example, we know that only 2 lines will render since x: 3 only exists in the domain